etsin_finder/frontend/js/components/dataset/formatChanger.jsx

```diff
@@ -0,0 +1,128 @@
+{
+  /**
+   * This file is part of the Etsin service
+   *
+   * Copyright 2017-2018 Ministry of Education and Culture, Finland
+   *
+   *
+   * @author    CSC - IT Center for Science Ltd., Espoo Finland <servicedesk@csc.fi>
+   * @license   MIT
+   */
+  }
+
+  import React, { Component } from 'react'
+  import PropTypes from 'prop-types'
+  import { withRouter } from 'react-router-dom'
+  import { withTheme } from 'styled-components'
+  import ErrorPage from '../errorpage'
+  import DatasetQuery from '../../stores/view/datasetquery'
+  import Accessibility from '../../stores/view/accessibility'
+
+
+  import FormatSelect from './formatselect'
+
+
+  class FormatChanger extends Component {
+    constructor(props) {
+      super(props)
+
+      this.state = {
+        formats: [],
+        error: false,
+        selected: '',
+      }
+
+      this.query = this.query.bind(this)
+    }
+
+    componentDidMount() {
```

Comment:
- There is no need to reset focus to this component, the focus is already reset to the page in `/components/dataset/index.js` that renders this component.
- The query is also done in the parent component so this is unnecessary. Now every time the dataset page is refreshed the same query is run in the parent and in this component.